### PR TITLE
Add embed post link to REST API

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Follow up from https://github.com/google/web-stories-wp/pull/8268. 

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by swissspidy** [comment]
Let's not call this `Confirm Link`.

The question in the editor is "Would you like to include it on a new post?" and yes, when the user confirms that question, they will navigate to that URL, but without that context this term doesn't make much sense.

This link navigates to the post block editor with a Web Stories block pre-filled to embed the story. That's what we should describe here.

So maybe something like `Embed Post Edit Link` (`embedPostLink`)?

File: tests/phpunit/tests/REST_API/Stories_Controller.php [hunk 1]
```diff
@@ -208,6 +208,9 @@ public function test_get_item() {
 		$this->assertArrayHasKey( 'preview_link', $data );
 		$view_link = get_preview_post_link( $story );
 		$this->assertSame( $view_link, $data['preview_link'] );
+
+		$this->assertArrayHasKey( 'confirm_link', $data );
+		$this->assertContains( (string) $story, $data['confirm_link'] );
```
Comment (swissspidy) [inline tests/phpunit/tests/REST_API/Stories_Controller.php]:
```suggestion
		$this->assertArrayHasKey( 'embed_post_link', $data );
		$this->assertContains( (string) $story, $data['embed_post_link'] );
```

The second assertion could probably check for `post-new.php` / `?from-web-story` to be a bit more precise

**Review by swissspidy — COMMENTED** [review]
Just a failing test but otherwise looks good 👍

File: includes/REST_API/Stories_Controller.php [hunk 2]
```diff
@@ -115,6 +117,10 @@ public function prepare_item_for_response( $post, $request ) {
 			$data['preview_link'] = $view_link;
 		}
 
+		if ( in_array( 'embed_post_link', $fields, true ) && current_user_can( 'edit_posts' ) ) {
+			$data['embed_post_link'] = add_query_arg( [ 'from-web-story' => $post->ID ], admin_url( 'post-new.php' ) );
+		}
```
Comment (spacedmonkey) [inline includes/REST_API/Stories_Controller.php:122]: @swissspidy I made `embed_post_link` optional, if the user can not edit posts. Makes no sense to show this link then take then to a permission error page.

Comment (spacedmonkey) [reply includes/REST_API/Stories_Controller.php:122]: Screenshot 2021-07-13 at 11 38 41

This is what it looks like without a `embed_post_link`

Comment (swissspidy) [reply includes/REST_API/Stories_Controller.php:122]: Awesome, thank you!

**Review by swissspidy — APPROVED** [review]

**Pull request merged**